doc/dev/kdb-operations.md

```diff
@@ -0,0 +1,159 @@
+# Proposal: KDB Operations
+
+There are four main _operations_ in `libelektra-kdb`: `open`, `get`, `set` and `close`.
+For each of these there is a `kdb*` function the user calls to trigger the operation and plugins implement a `elektra<Plugin>*` function for each of the operations they support (at least `get`).
+
+Additionally, plugins may implement `elektra<Plugin>Commit` and `elektra<Plugin>Error`.
+These are part of the `set` operation and there is no corresponding `kdbCommit` or `kdbError` function available in `libelektra-kdb`.
+
+The operations `get` and `set` also have different _phases_:
+
+- The `get` operation has: `init`, `resolver`, `cachecheck`, `prestorage`, `storage` and `poststorage`.
+- The `set` operation has: `resolver`, `prestorage`, `storage` and `poststorage` followed by `precommit`, `commit` and `postcommit` if the previous phases where successful or by `prerollback`, `rollback` and `postrollback` if the previous phases failed.
+
+These phases are implemented by a backend plugin.
```

Comment:
The next line links to `backend-plugins.md`, which contains this line:

> For each of the phases of a `get` operation `libelektra-kdb` calls the backend plugin's `elektra<Plugin>Get` function once.

Also since the document is called `kdb-operations` and begins with

> There are four main _operations_ in `libelektra-kdb`: `open`, `get`, `set` and `close`.

I think the conclusion should be that this mainly involves `libelektra-kdb`, at least as far as this document is concerned.

Reply:
The art is to make a document clear from the beginning and not only after rereading several times.

Reply:
> The art is [...]

Well I'm an engineer and not an artist ;)

> clear from the beginning

1. if everything was perfect the first time, we wouldn't need reviews
2. this PR is still a draft, so it explicitly still a in WIP